File: exams/frmSubjectRankPrompt2.vb
Public Class frmSubjectRankPrompt2
    Private Sub frmSubjectRankPrompt2_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        radFirst.Checked = True
    End Sub

    Private Sub btnCancel_Click(sender As Object, e As EventArgs) Handles btnCancel.Click
        Me.Close()
    End Sub

    Private Sub btnAnalyze_Click(sender As Object, e As EventArgs) Handles btnAnalyze.Click
        If Not IsNumeric(txtNumber.Text) Then
            If radNone.Checked = True Then
                filterType = "None"
            Else
                MsgBox("Please enter the result limit or the value is not a number")
                Return
            End If
        Else
            rankno = txtNumber.Text
            If radFirst.Checked = True Then
                filterType = "Top"
            ElseIf radLast.Checked = True Then
                filterType = "Bottom"
            Else
                filterType = "None"
            End If

        End If

        Dim bestSubject As New frmBestStudentSubject
        bestSubject.ShowDialog()

        Me.Close()

    End Sub
End Class

File: exams/frmSubjectRankPrompt2.cs
using System;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace exams
{
    public partial class frmSubjectRankPrompt2
    {
        public frmSubjectRankPrompt2()
        {
            InitializeComponent();
            _btnCancel.Name = "btnCancel";
            _btnAnalyze.Name = "btnAnalyze";
        }

        private void frmSubjectRankPrompt2_Load(object sender, EventArgs e)
        {
            radFirst.Checked = true;
        }

        private void btnCancel_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void btnAnalyze_Click(object sender, EventArgs e)
        {
            if (!Information.IsNumeric(txtNumber.Text))
            {
                if (radNone.Checked == true)
                {
                    publicSubsNFunctions.filterType = "None";
                }
                else
                {
                    Interaction.MsgBox("Please enter the result limit or the value is not a number");
                    return;
                }
            }
            else
            {
                publicSubsNFunctions.rankno = Conversions.ToInteger(txtNumber.Text);
                if (radFirst.Checked == true)
                {
                    publicSubsNFunctions.filterType = "Top";
                }
                else if (radLast.Checked == true)
                {
                    publicSubsNFunctions.filterType = "Bottom";
                }
                else
                {
                    publicSubsNFunctions.filterType = "None";
                }
            }

            var bestSubject = new frmBestStudentSubject();
            bestSubject.ShowDialog();
            Close();
        }
    }
}